lib/linguist/heuristics.rb

```diff
@@ -223,5 +223,30 @@ def call(data)
         Language["Text"]
       end
     end
+    disambiguate "PLSQL", "SQLPL", "PLpgSQL", "SQL" do |data|
+      match = nil 
+      #unique keywords for each language
+      plpgsqlkeywords = [/^\\i\b/i, /begin ?(work|transaction)?;/i , /AS \$\$/i, /RAISE EXCEPTION/i, /LANGUAGE plpgsql\b/i ]
+      plsqlkeywords = [ /pragma\b/i , /constructor\W+function\b/i]
+      #sqlplkeywords = [  ]
+
+      #common keywords that are not SQL
+      plkeywords = [/begin\b/i,/boolean\b/i, /package\b/i,/exception\b/i, /^end;\b/i ]
+
+      re = Regexp.union(plpgsqlkeywords)
+      if data.match(re)
```

Comment:
It would be great to see this written more like the current heuristics. Take a look at [this one](https://github.com/github/linguist/blob/7a141a923c4d0aefc0f59e888cf585f6eb253a60/lib/linguist/heuristics.rb#L145-L151) for `Scala`:

```ruby
disambiguate "Scala", "SuperCollider" do |data|
  if /\^(this|super)\./.match(data) || /^\s*(\+|\*)\s*\w+\s*{/.match(data) || /^\s*~\w+\s*=\./.match(data)
    Language["SuperCollider"]
  elsif /^\s*import (scala|java)\./.match(data) || /^\s*val\s+\w+\s*=/.match(data) || /^\s*class\b/.match(data)
    Language["Scala"]
  end
end
```